src/preamble.js

```diff
@@ -1071,14 +1084,22 @@ function createWasm(env) {
         typeof WebAssembly.instantiateStreaming === 'function' &&
         !isDataURI(wasmBinaryFile) &&
         typeof fetch === 'function') {
-      return WebAssembly.instantiateStreaming(fetch(wasmBinaryFile, { credentials: 'same-origin' }), info)
-        .then(receiveInstantiatedSource, function(reason) {
-          // We expect the most common failure cause to be a bad MIME type for the binary,
-          // in which case falling back to ArrayBuffer instantiation should work.
-          err('wasm streaming compile failed: ' + reason);
-          err('falling back to ArrayBuffer instantiation');
-          instantiateArrayBuffer(receiveInstantiatedSource);
+      fetch(wasmBinaryFile, { credentials: 'same-origin' }).then(function (response) {
+#if 'emscripten_generate_pc' in addedLibraryItems
+        response.clone().arrayBuffer().then(function (buffer) {
```

Comment:
I understand the async stuff now I think, thanks.

So this clones the response, and we in effect are doing two requests, one for streaming compilation, and one for the binary? This seems ok enough for sanitizers I suppose. Please add a comment though, that mentions the duplicate request is not optimal, but we think it's ok since this isn't for normal production builds etc.

Reply:
Well, it's like, two virtual requests. I checked in the browser console and it doesn't actually do a second request. I'll make a note of this.